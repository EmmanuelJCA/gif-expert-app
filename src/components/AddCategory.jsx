import { useState } from "react"

const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const handleSubmit = ( event ) => {
        event.preventDefault()

        if( inputValue.trim().length <= 1 ) return

        setInputValue('')
        onNewCategory( inputValue.trim() )
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
            type="text" 
            placeholder="Buscar gif"
            value={ inputValue }
            onChange={ handleInputChange }
            />
        </form>
    )
}

export default AddCategory