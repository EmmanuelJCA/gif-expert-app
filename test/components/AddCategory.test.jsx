import { fireEvent, render, screen } from '@testing-library/react'
import { default as AddCategory} from '../../src/components/AddCategory'

describe('Tests on <AddCategory />', () => {

    test('should change the value in the text box', () => {

        render( 
            <AddCategory onNewCategory={() => {}} />
        )
        const input = screen.getByRole('textbox')

        fireEvent.input( input, { 
                target: { value: 'Saitama'} 
            } 
        )

        expect( input.value ).toBe('Saitama')

    })

})