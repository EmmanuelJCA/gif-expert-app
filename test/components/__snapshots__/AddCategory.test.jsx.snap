// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should match the snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
